games/virus_plowing.js

```diff
@@ -0,0 +1,235 @@
+/*
+@title: virus_plowing
+@author: Danush Ramanan
+*/
+
+/*
+CONTROLS:
+  W - Move Up
+  S - Move Down
+  A - Move Left
+  D - Move Right
+
+  I - Info
+*/
+
+class Player {
+  constructor(x, y) {
+    this.x = x;
+    this.y = y;
+  }
+
+  move(dir) {
+    switch(dir) {
+      case 'w':
+        this.y--;
+        break;
+      case 'a':
+        this.x--;
+        break;
+      case 's':
+        this.y++;
+        break;
+      case 'd':
+        this.x++;
+        break;
+      default:
+        break;
+    }
+  }
+
+  update() {
+    getFirst(playerKey).x = this.x;
+    getFirst(playerKey).y = this.y;
+  }
+}
+
+const playerKey = "p";
+const virusKey = "v";
+const shieldKey = "s";
+
+const killVirusSound = tune`
+124.48132780082987: c5~124.48132780082987,
+124.48132780082987: f5~124.48132780082987 + c5-124.48132780082987,
+124.48132780082987: f5-124.48132780082987,
+3609.9585062240662`;
+const dieSound = tune`
+205.4794520547945: f5-205.4794520547945 + b4^205.4794520547945,
+205.4794520547945: f5-205.4794520547945 + b4^205.4794520547945,
+205.4794520547945: c5-205.4794520547945 + f4^205.4794520547945,
+205.4794520547945: c5-205.4794520547945 + f4^205.4794520547945,
+205.4794520547945: e5-205.4794520547945 + a4^205.4794520547945,
+205.4794520547945: d5-205.4794520547945 + g4^205.4794520547945,
+205.4794520547945,
+205.4794520547945: b4/205.4794520547945,
+205.4794520547945: b4/205.4794520547945,
+4726.027397260274`;
+
+let score = 0;
+let showingInstructions = false;
+
+setLegend(
+  [ playerKey, bitmap`
+3333333333333333
+3333333333333333
+3333333333333333
+3330003333000333
+3330003333000333
+3333333333333333
+3333333333333333
+3333333333333333
+3333333333333333
+3333333333333333
+3333333333333333
+...33......33...
+...33......33...
+...33......33...
+...33......33...
+..3333....3333..`],
+  [ virusKey, bitmap`
+4444444444444444
+4444444444444444
+4444444444444444
+4444444444444444
+4444444444444444
+4444444444444444
+4444444444444444
+4444444444444444
+4444444444444444
+4444444444444444
+4444444444444444
+4444444444444444
+4444444444444444
+4444444444444444
+4444444444444444
+4444444444444444`],
+  [ shieldKey, bitmap`
+................
+................
+7......77......7
+7777777777777777
+7777777777777777
+7777779779777777
+7777999999997777
+7777944444497777
+7777794444977777
+7777779449777777
+.77777799777777.
+.77777777777777.
+..777777777777..
+..777777777777..
+...7777777777...
+................` ]
+);
+
+setSolids([playerKey, shieldKey]);
+
+let level = 0;
+const levels = [
+  map`
+..................
+.p................
+...............v..
+..................
+..................
+..................
+..................
+..................
+.......s..........
+..................
+..................`,
+  map`
+..................
+..................
+..................
+..................
+..................
+..................
+..................
+..................
+..................
+..................
+..................`,
```

Comment:
what is the function of this empty map?

Reply:
The function of the empty map is for the game over screen

Reply:
oh, right!